File: 1646-KthMissingPositiveNumber/1646-KthMissingPositiveNumber.py
# Last updated: 6/17/2025, 10:25:39 PM
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        for i in range(len(arr)):
            missing = arr[i] - (i + 1)
            if missing >= k:
                return k + i
        return k + len(arr)

        

File: 1646-KthMissingPositiveNumber/1646-KthMissingPositiveNumber.java
// Last updated: 8/23/2025, 6:49:40 PM
class Solution {
    public int findKthPositive(int[] arr, int k) {
        for(int i = 0; i<arr.length; i++){
            if(arr[i] <= k){
                k++;
            }
            else{
                break;
            }
        }
        return k;

    }
}